cryptoprotocol/src/main/java/de/upb/crypto/incentive/cryptoprotocol/model/Provider.java

```diff
@@ -0,0 +1,41 @@
+package de.upb.crypto.incentive.cryptoprotocol.model;
+
+import de.upb.crypto.incentive.cryptoprotocol.model.keys.provider.ProviderKeyPair;
+import de.upb.crypto.math.serialization.StringRepresentation;
+import lombok.Data;
+
+@Data
+public class Provider
+{
+    private ProviderKeyPair providerKeyPair;
+
+    /**
+     * effectively a no args constructor, generating a new key pair. Used for starting a new session.
+     * @param pp public parameters
+     */
+    public Provider(PublicParameters pp)
+    {
+        this.providerKeyPair = Setup.providerKeyGen(pp);
+    }
+
+    /**
+     * constructor taking "dumb string", for resuming previous session (e.g. after closing the provider app)
+     * @param serializedProviderKeyPair
+     */
+    public Provider(String serializedProviderKeyPair)
```

Comment:
Why don't pass serialized pp as well? We could add a superclass for user and provider pp for this

Reply:
Added pp as a trust anchor (provided via constructor and stored only there) for the IncentiveSystem class